### PR TITLE
Add hook scripts to genezio local command

### PR DESCRIPTION
**Pull request by vladiulianbogdan**
# Pull Request Template

## Type of change

- [x] 🍕 New feature

## Description

Add hook scripts to genezio local command. This allows the user to automate the execution of some scripts before running `genezio local`. For example, one useful usage of these scripts is to execute `npm install` and `npm link <sdk>` when `genezio local` is executed.

I've added to hook events: `preStartLocal` and `preReloadLocal`.

### CODE REVIEW

File: src/telemetry/telemetry.ts [hunk 1]
```diff
@@ -20,6 +20,8 @@ export enum TelemetryEventTypes {
     GENEZIO_GENERATE_SDK_ERROR = "GENEZIO_GENERATE_SDK_ERROR",
     GENEZIO_ADD_CLASS = "GENEZIO_ADD_CLASS",
     GENEZIO_DEPLOY_ERROR = "GENEZIO_DEPLOY_ERROR",
+    GENEZIO_PRE_START_LOCAL_SCRIPT = "GENEZIO_PRE_START_LOCAL_SCRIPT",
+    GENEZIO_PRE_RELOAD_LOCAL_SCRIPT = "GENEZIO_PRE_RELOAD_LOCAL_SCRIPT",
```
Comment (costinsin) [inline src/telemetry/telemetry.ts:24]: Both of these EventTypes are used on the error side of running the scripts. Please add a `_ERROR` prefix to the event names.

File: src/commands/local.ts [hunk 2]
```diff
@@ -141,17 +143,46 @@ export async function startLocalEnvironment(options: GenezioLocalOptions) {
         eventType: TelemetryEventTypes.GENEZIO_LOCAL,
         commandOptions: JSON.stringify(options),
     });
+    let yamlProjectConfiguration = await getProjectConfiguration();
+
+    if (yamlProjectConfiguration.scripts?.preStartLocal) {
+        log.info("Running startLocal script...");
```
Comment (costinsin) [inline src/commands/local.ts:149]:
```suggestion
        log.info("Running preStartLocal script...");
```

File: src/commands/local.ts [hunk 3]
```diff
@@ -141,17 +143,46 @@ export async function startLocalEnvironment(options: GenezioLocalOptions) {
...
-        const yamlProjectConfiguration = await getProjectConfiguration();
+        yamlProjectConfiguration = await getProjectConfiguration();
         if (!Language[yamlProjectConfiguration.language as keyof typeof Language]) {
             log.info(
                 "This sdk.language is not supported by default. It will be treated as a custom language.",
             );
         }
 
+        if (yamlProjectConfiguration.scripts?.preReloadLocal) {
+            log.info("Running reloadLocal script...");
```
Comment (costinsin) [inline src/commands/local.ts:173]:
```suggestion
            log.info("Running preReloadLocal script...");
```

**Review by costinsin — APPROVED** [review]
[looksgood](https://app.graphite.dev/macros/Genez-io)

LGTM!

File: src/models/yamlProjectConfiguration.ts [hunk 4]
```diff
@@ -197,17 +197,23 @@ export class YamlScriptsConfiguration {
     postBackendDeploy?: string;
     postFrontendDeploy?: string;
     preFrontendDeploy?: string;
+    preStartLocal?: string;
+    preReloadLocal?: string;
```
Comment (stefandarius) [inline src/models/yamlProjectConfiguration.ts:201]: Shouldn't we have a `postStartLocal` hook? For example, the `npm link <SDK>` command wouldn't work in a `preStartLocal` script, as the SDK is not yet generated.

Comment (vladiulianbogdan) [reply src/models/yamlProjectConfiguration.ts:201]: Correct.

**Review by stefandarius — APPROVED** [review]

**Pull request merged**